src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -505,8 +512,10 @@ record Element(Label target, Label next, Object label) {}
                                                         MethodType.methodType(int.class,
                                                                               Object.class,
                                                                               int.class,
-                                                                              BiFunction.class));
-            return MethodHandles.insertArguments(typeSwitch, 2, new ResolvedEnumLabels(caller, enumDescs.toArray(s -> new EnumDesc<?>[s])));
+                                                                              BiPredicate.class,
+                                                                              Class[].class));
+            return MethodHandles.insertArguments(typeSwitch, 2, new ResolvedEnumLabels(caller, enumDescs.toArray(s -> new EnumDesc<?>[s])),
```

Comment:
you can use the method reference `EnumDesc[]::new`instead of `s -> new EnumDesc<?>[s]` and same below Class[]::new  (the wirldcard should not be necessary)

Reply:
Done, thanks.